### PR TITLE
feat: add interfaces for cloud pubsub publishers and subscribers

### PR DESCRIPTION
**Pull request by dpcollins-google**

### CODE REVIEW

File: google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/v1/SubscriberInterface.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.pubsub.v1;
+
+import com.google.api.core.ApiService;
+
+// The core interface for a Cloud Pub/Sub subscriber, consisting only of ApiService methods.
```
Comment (hannahrogers-google) [inline google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/v1/SubscriberInterface.java:21]: Can we change this to be a java doc comment
/*
 */

And add a link to subscriber documentation, like you did with the PublisherInterface

**Comment by hannahrogers-google** [comment]
You can also resolve the code format failure by running 'mvn clean verify'

File: google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/v1/SubscriberInterface.java (hunk 1 above)
Comment (dpcollins-google) [reply google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/v1/SubscriberInterface.java]: Done.

File: google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/v1/SubscriberInterface.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.pubsub.v1;
+
+import com.google.api.core.ApiService;
+
+/** The core interface for a Cloud Pub/Sub
```
Comment (hannahrogers-google) [inline google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/v1/SubscriberInterface.java:21]: 'The core interface...' should be on a new line

Running 'mvn clean verify' should resolve this

Comment (dpcollins-google) [reply google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/v1/SubscriberInterface.java]: Done.

**Pull request merged**